paymentsheet/src/test/java/com/stripe/android/customersheet/state/DefaultCustomerSheetLoaderTest.kt

```diff
@@ -407,6 +411,41 @@ class DefaultCustomerSheetLoaderTest {
         )
     }
 
+    @OptIn(ExperimentalCardBrandFilteringApi::class)
+    @Test
+    fun `when there are saved cards with disallowed brands they are filtered out`() = runTest {
+        val loader = createCustomerSheetLoader(
+            paymentMethods = listOf(
+                PaymentMethodFactory.card(id = "pm_12345").update(
+                    last4 = "1001",
+                    addCbcNetworks = false,
+                    brand = CardBrand.Visa,
+                ),
+                PaymentMethodFactory.card(id = "pm_123456").update(
+                    last4 = "1000",
+                    addCbcNetworks = false,
+                    brand = CardBrand.AmericanExpress,
+                )
+            ),
+            savedSelection = null,
+        )
+
+        val config = CustomerSheet.Configuration(
+            merchantDisplayName = "Example",
+            cardBrandAcceptance = PaymentSheet.CardBrandAcceptance.disallowed(
+                listOf(PaymentSheet.CardBrandAcceptance.BrandCategory.Visa)
+            )
+        )
+
+        val state = loader.load(config).getOrThrow()
+        assertThat(state.customerPaymentMethods.count()).isEqualTo(
+            1
+        )
+        assertThat(state.customerPaymentMethods.first().card?.brand ?: CardBrand.Unknown).isEqualTo(
```

Comment:
```suggestion
        assertThat(state.customerPaymentMethods.first().card?.brand).isEqualTo(
```

This will fail if the brand is null

Reply:
Good call, removed this in the PaymentSheetLoader test too!